### PR TITLE
Use N-1 snapshotID to propose

### PR DESCRIPTION
**Pull request by preethamr**
## Description

Use N-1 snapshotID to propose

## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [x] Refactoring
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/agents/lighthouse/src/tasks/propose/errors.ts [hunk 1]
```diff
@@ -131,18 +131,3 @@ export class AggregateRootChecksFailed extends NxtpError {
     });
   }
 }
-
-export class SubgraphDelayed extends NxtpError {
```
Comment (LayneHaber) [inline packages/agents/lighthouse/src/tasks/propose/errors.ts:135]: are we sure we want to remove this error completely?

what happens when carto is hours behind?

File: packages/agents/lighthouse/src/tasks/propose/operations/proposeHub.ts [hunk 2]
```diff
@@ -102,57 +101,63 @@ export const proposeHub = async () => {
...
-            });
-            await database.saveSnapshotRoots([
-              {
-                id: latestSnapshotId.toString(),
-                spokeDomain: +domain,
-                root: latestSnapshotRoot,
+          const outboundRootTimestamp = await database.getOutboundRootTimestamp(domain, domainOutboundRoot);
+          if (outboundRootTimestamp && outboundRootTimestamp < latestSnapshotTimestamp) {
+            const messageRootCount = await database.getMessageRootCount(domain, domainOutboundRoot);
+            if (messageRootCount) {
```
Comment (LayneHaber) [inline packages/agents/lighthouse/src/tasks/propose/operations/proposeHub.ts:110]: do we still need to store the virtual roots?

File: packages/agents/lighthouse/src/tasks/propose/operations/proposeHub.ts [hunk 3]
```diff
@@ -102,57 +101,63 @@ export const proposeHub = async () => {
...
+              logger.debug("Storing the virtual snapshot root in the db", requestContext, methodContext, {
+                domain,
                 count: messageRootCount + 1,
-                timestamp: latestSnapshotTimestamp,
+                snapshotId: latestSnapshotId + 1, // We are using N - 1 snapshot ID
+              });
+              await database.saveSnapshotRoots([
+                {
+                  id: latestSnapshotId.toString(),
```
Comment (LayneHaber) [inline packages/agents/lighthouse/src/tasks/propose/operations/proposeHub.ts:118]: does this need to be incremented?

**Comment by preethamr** [comment]
Closing this as not valid anymore.

**Pull request closed without merging**